src/ipc/ipc4/helper.c

```diff
@@ -755,3 +755,22 @@ int ipc4_add_comp_dev(struct comp_dev *dev)
 
 	return IPC4_SUCCESS;
 };
+
+int ipc4_find_dma_config(struct ipc_config_dai *dai, uint8_t *data_buffer, uint32_t size)
+{
+#if defined(CONFIG_ACE_VERSION_2_0)
+	struct ipc_dma_config *dma_config =
+	    (struct ipc_dma_config *)(data_buffer + (size * 4) - sizeof(struct ipc_dma_config));
+	uint32_t *dma_config_id =
+	    (uint32_t *)(data_buffer + (size * 4) - sizeof(struct ipc_dma_config) - 8);
```

Comment:
looks like we are getting some structures from a data blob. Its probably better to have these as static inlines or macros so that they can be reused by others (and mean less chance of others introducing errors)

Reply:
Actually this is behind the interface specific blob. It was added at the end probably to keep backward compatibility and use the same IPC for all generations. 
Changed to macros.